### ISSUE TITLE
Better examples in documentation for @angular/upgrade/static

### ISSUE THREAD

**Issue by Mnaaz** [angular/angular#25928]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior
I have been looking for code examples using @angular/upgrade/static to upgrade/downgrade AngularJS to Angular and vice versa. The documentation does provide some code samples, but they are not very understandable in my opinion. The documentation for downgradeComponent for instance, does not provide any examples for downgrading to an AngularJS application written in JavaScript. All examples have an AngularJS app written in TypeScript.

## Expected behavior
The documentation for @angular/upgrade/static should contain understandable examples for the following Components:
- UpgradeModule
- UpgradeComponent
- downgradeModule
- downgradeComponent
- downgradeInjectable

Specifically, examples for downgrading to AngularJS written in JavaScript. Working examples should be provided in [GitHub](https://github.com/angular/angular/tree/master/aio/content/examples/upgrade-module) with a link in the documentation.

## Minimal reproduction of the problem with instructions
Visit https://angular.io/api/upgrade/static

## What is the motivation / use case for changing the behavior?
Complete and working examples in the documentation make it easier to build hybrid apps and to interchange Angular with AngularJS if you have large AngularJS apps and cannot migrate to Angular that easily.

**Comment by gkalpak** [comment]
Related to #19102.

**Comment by jenniferfell** [comment]
Minimally, we need good links from the upgrade/static API ref pages to the upgrade guide: https://angular.io/guide/upgrade

**Comment by jenniferfell** [comment]
DowngradeModule is discussed in https://angular.io/guide/upgrade-performance

**Comment by Mnaaz** [comment]
Thanx @jenniferfell . I will look into the forum. Full examples in the documentation and working ones in GitHub would be awesome.

**Comment by kegluneq** [comment]
[Bootstrapping hybrid applications](https://angular.io/guide/upgrade#bootstrapping-hybrid-applications) also suggests using `platformBrowserDynamic`. IIUC static is now the recommended approach

**Comment by gkalpak** [comment]
You can use both. (Static is preferred for production.)
How to use static is explained [further below][1]. 

[1]: https://angular.io/guide/upgrade#using-ahead-of-time-compilation-with-hybrid-apps

**Comment by ecureuilpop** [comment]
Hi, I have found this "closed" issue (I don't ever know how I reached that page... I were just trying to find something helpful to go througth the downgrading Angular Component), I'm kind of desesperate now.

I'm trying to use a really simple Angular component in my hybride AngularJS app, and I'm not really good to achiving this task. In fact, I was so much happy to see my downgraded HelloWord component (after 3 days spending googling) in my AngularJS app, so I decided to follow my expirience with an evoluated  "Hello World" component that does "*ngIf" so depends on the "commonModule". As result,  I'm stuck with the same error for multiples days, and googling dosen't help.
```
scripts.js:57488 Error: StaticInjectorError(AppModule)[NgForOf -> ViewContainerRef]: 
  StaticInjectorError(Platform: core)[NgForOf -> ViewContainerRef]: 
    NullInjectorError: No provider for ViewContainerRef!
    at NullInjector.push../node_modules/@angular/core/fesm5/core.js.NullInjector.get (vendor.js:147297)
    at resolveToken (vendor.js:147542)
    at tryResolveToken (vendor.js:147486)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (vendor.js:147383)
    at resolveToken (vendor.js:147542)
    at tryResolveToken (vendor.js:147486)
    at StaticInjector.push../node_modules/@angular/core/fesm5/core.js.StaticInjector.get (vendor.js:147383)
    at resolveNgModuleDep (vendor.js:159619)
    at NgModuleRef_.push../node_modules/@angular/core/fesm5/core.js.NgModuleRef_.get (vendor.js:160308)
    at resolveDep (vendor.js:160679) "<hello-world class="ng-scope">"
```

 Could you please suggest any doc, any sample that can help me please. Thank you.

**Comment by ardunster** [comment]
Rather than opening a new issue I will add my voice to this mostly relevant issue. 

I just spent 2-3 days digging on google trying to resolve an error `Can't resolve all parameters for AppModule: (?)` on an attempt to add Angular12 to my existing AngularJS project that I am manually compiling via webpack. (I also did a side foray into trying to use the AngularCLI, however that seemed to think that every time I added `app` to a source path I obviously was confused and didn't mean to and conveniently removed it for me - and then for some reason could never find my files in `app/src/filename.ts`. I never did find any way to change this behavior and eventually gave up.)  

Ultimately the resolution was to add `import 'reflect-metadata'` to my main.ts file.  It would be great if the documentation could include a list of everything that actually needs to be imported if you're doing this manually - as many people who are trying to upgrade an existing project will be.

**Comment by petebacondarwin** [comment]
@ardunster I would strongly encourage you to build at least the Angular 12 parts of the application with the Angular CLI. It does a lot for you, especially in terms of consuming libraries and localization, as well as optmisation of the final output, which is very hard to implement in a custom webpack setup.

I am not sure why you needed to add `import 'reflect-metadata';` - perhaps because you are relying upon JIT compilation of the Angular parts of the code (since you are not using the Angular CLI)?

**Comment by ardunster** [comment]
No idea, but it didn't work without it no matter what I did, and it did work with it. I tried using the Angular CLI. I fought with it for a day or so. It could never find `app/src/index.ts` no matter what I put in root and sourceroot, if it got that far at all it *always* dropped the `app` even when I explicitly put it in the path for index.ts itself (so instead of looking in `rootfolder/app/src/` it would *always* look in `rootfolder/src/` for it..which doesn't exist.)  Documentation did nothing to explain which paths were supposed to be relative to what and makes too many assumptions about what you're supposed to know about what these terms mean, and I couldn't find any way to make it output debug information so I could see what logic it was actually attempting, and I don't have time to keep fighting with a tool that refuses to do what I tell it to, so I went back to webpack. Maybe I'll try again some day.
